calibration_assistant.cpp

```diff
@@ -0,0 +1,1250 @@
+                }
+            }
+            else
+                ShowExplanationMsg(dec);
+        }
+    }
+    else
+    {
+        ShowError(_("Mount can't report its pointing position"), true);
+        if (m_pTimer)
+            m_pTimer->Stop();
+    }
+}
+// Return of 'true' means error occurred
+bool CalibrationAssistant::GetCalibPositionRecommendations(int* HA, int* Dec) const
+{
+    double hourAngle;
+    int bestDec;
+    int bestOffset;
+    double lst;
+    double ra;
+    double dec;
+    double eqAltitude;
+    bool pointingEast = false;
+    bool errorSeen = false;
+
+    bestDec = defBestDec;
+    bestOffset = defBestOffset;
+    try
+    {
+        if (pPointingSource && pPointingSource->CanReportPosition())
+        {
+            if (pPointingSource->PreparePositionInteractive())
+                return true;
+            if (pPointingSource->GetCoordinates(&ra, &dec, &lst))
+            {
+                throw ERROR_INFO("CalPositionRecommendations: Mount not reporting pointing position");
+            }
+            hourAngle = norm(lst - ra, -12.0, 12.0);
+            if (hourAngle <= 0)
+                bestOffset = -bestOffset;
+            // Check that we aren't pointing down in the weeds at the default Dec location
+            double lat, lon;
+            if (pPointingSource->GetSiteLatLong(&lat, &lon))
+                bestDec = 0;
+            else if (lat >= 0)
+            {
+                eqAltitude = 90.0 - lat;
+                if (eqAltitude < 30)        // far north location
+                    bestDec += 30 - eqAltitude;
+            }
+            else
+            {
+                eqAltitude = 90 + lat;
+                if (eqAltitude < 30)        // far south location
+                    bestDec -= (30 - eqAltitude);
+            }
+            *HA = bestOffset;
+            *Dec = bestDec;
+            m_pSlewBtn->Enable(pPointingSource->CanSlew());
+        }
+        else
+        {
+            throw ERROR_INFO("CalPositionRecommendations: mount not connected or not reporting pointing info");
+        }
+    }
+    catch (const wxString& msg)
+    {
+        POSSIBLY_UNUSED(msg);
+        *HA = defBestOffset;
+        *Dec = defBestDec;
+        errorSeen = true;
+    }
+    return errorSeen;
+}
+
+void CalibrationAssistant::ShowExplanationMsg(double dec)
+{
+    wxString slewCond;
+    if (pPointingSource->CanSlew())
+        slewCond = _("Use the 'slew' button to move the scope as close as possible to Dec = 0. ");
+    else
+        slewCond = _("Slew the scope as close as possible to Dec = 0. ");
+    if (fabs(dec) > 80)
+    {
+        m_pExplanation->SetLabelText(_("Calibration is likely to fail this close to the pole.\n") + slewCond);
+    }
+    else if (fabs(dec) > degrees(Scope::DEC_COMP_LIMIT))
+    {
+        m_pExplanation->SetLabelText(_("Declination compensation will not work if you calibrate within 30 degrees of the pole.\n") + slewCond);

```

Comment:
I guess this is a pre-existing message but I think this message is confusing for newbs who won't know what dec comp is.  Maybe we should be more direct and just tell them to calibrate closer to dec 0, or to disable dec comp?